pilot/pkg/xds/delta.go

```diff
@@ -514,25 +510,24 @@ func (s *DiscoveryServer) pushDeltaXds(con *Connection, w *model.WatchedResource
 		Nonce:             nonce(req.Push.PushVersion),
 		Resources:         res,
 	}
-	currentResources := slices.Map(res, func(r *discovery.Resource) string {
+	currentResources := sets.New(slices.Map(res, func(r *discovery.Resource) string {
```

Comment:
BUt this function `pushDeltaXds` is frequently called during each push, doesnot it comprise the saving in `deltaWatchedResources`

Reply:
I've pushed up a commit that optimizes this more and only makes a Set when it is required. I will compare that code vs the existing code.

case 1: usedDelta (old L520): no change
case 2: req.Full (old L522): before we make a set of all resources and then delete some. Now we do basically the same, but do not make a currentResources slice so its slightly more efficient
case 3: shouldSetWatchedResources && usedDelta (L531): before we make a new set of all resources, then do some adds and removes. Now we make a new set and do the same (but no need to construct currentResources)
case 4: no changes


so I think with the new commit I pushed this is strictly more efficient